packages/foam-vscode/src/features/preview/wikilink-embed.ts

```diff
@@ -194,7 +193,7 @@ function contentExtractor(
   parser: ResourceParser,
   workspace: FoamWorkspace
 ): string {
-  let noteText = readFileSync(note.uri.toFsPath()).toString();
+  let noteText = vsWorkspace.fs.readFile(toVsCodeUri(note.uri)).toString();
```

Comment:
I think `vsWorkspace.fs.readFile` returns a promise, this was to me the only open question as to how to transition in the code

Reply:
Hi, is this still open? If yes can we get around by writing a new `markdown-it` plugin to handle the async operation?
Their documentation has some [guidelines](https://github.com/markdown-it/markdown-it/blob/master/docs/development.md#i-need-async-rule-how-to-do-it). Though I don't follow the wording aptly but I can give it a quick stab by forking `markdownItRegex`. WDYT?

Reply:
This is still open, you can find all the info here: https://github.com/foambubble/foam/pull/1295